src/librustc_middle/hir/map/mod.rs

```diff
@@ -1087,8 +1085,8 @@ fn hir_id_to_string(map: &Map<'_>, id: HirId) -> String {
 
 pub fn provide(providers: &mut Providers<'_>) {
     providers.def_kind = |tcx, def_id| {
-        if let Some(hir_id) = tcx.hir().as_local_hir_id(def_id) {
-            tcx.hir().def_kind(hir_id)
+        if let Some(def_id) = def_id.as_local() {
+            tcx.hir().def_kind(tcx.hir().as_local_hir_id(def_id))
         } else {
             bug!("calling local def_kind query provider for upstream DefId: {:?}", def_id);
```

Comment:
For query providers you can probably just use `.expect_local()`. Maybe we should eventually make it possible to take `LocalDefId` for a query keyed on `DefId` but that's probably too complicated.